butterfly-slack/src/main/java/com/paypal/butterfly/slack/SlackNotifierConfig.java

```diff
@@ -0,0 +1,140 @@
+                    String postAbortPath = new File(this.getClass().getResource("/postAbort.json").toURI()).getAbsolutePath();
+                    Client client = ClientBuilder.newClient();
+                    target = client.target("https://www.slack.com/api/");
+                    token = System.getProperty("slack.token");
+                    postSuccessJson = new String(Files.readAllBytes(Paths.get(postSuccessPath)));
+                    postAbortJson = new String(Files.readAllBytes(Paths.get(postAbortPath)));
+                    //TODO get butterflyUserID, details link and github url from Pupa
+                    butterflyUserIDs = new String[]{"edwardhwang@berkeley.edu", "ehwang0824@gmail.com"};
+                } catch(Exception ex) {
+                    logger.error("Unable to initialize slack notifier", ex);
+                }
+            }
+
+            @Override
+            public void postTransformation(TransformationRequest transformationRequest, TransformationResult transformationResult) {
+                postNotification(postSuccessJson);
+            }
+
+            @Override
+            public void postTransformationAbort(TransformationRequest transformationRequest, TransformationResult transformationResult) {
+                postNotification(postAbortJson);
+            }
+
+
+            private void postNotification(String jsonMessage){
+                for (String id : butterflyUserIDs) {
+                    //TODO: Change hard coded email into butterflyUserIDs
+                    String email = id + ""; //id + "@paypal.com";
+                    String slackUserID = getSlackUserID(target, token, email);
+                    String slackChannelID = getSlackChannelID(target, token, slackUserID);
+                    postSlackMessage(target, token, jsonMessage, slackChannelID);
+                }
+            }
+
+            /**
+             * GET slack user IDs from PayPal email using the users.lookupByEmail web api call
+             */
+            private String getSlackUserID(WebTarget target, String token, String email){
+                    JsonObject slackResponse = new JsonParser().parse(target
+                            .path("users.lookupByEmail")
+                            .queryParam("email", email)
+                            .request(MediaType.APPLICATION_JSON)
+                            .header("Authorization", "Bearer " + token)
+                            .get(String.class))
+                            .getAsJsonObject();
+                    String slackUserID = "";
+                    try {
+                        slackUserID = slackResponse.getAsJsonObject("user").get("id").getAsString();
+                    } catch (Exception e) {
+                        logger.warn("An exception happened when retrieving the slack user id " + slackResponse.get("error").getAsString(), e);
```

Comment:
You are catching `Exception`, which is very broad. Consider catching more specific exceptions here. Also, make sure `slackResponse` is not `null` if you decide to use it.